bin/preprocess.ml

```diff
@@ -0,0 +1,82 @@
+open Stdune
+open Import
+
+let doc = "Print the preprocessed output"
+
+let man =
+  [ `S "DESCRIPTION"
+  ; `P
+      {|$(b,dune preprocess FILE) build the given FILE and print the preprocessed output|}
+  ; `Blocks Common.help_secs
+  ]
+
+let info = Term.info "preprocess" ~doc ~man
+
+let change_ext f path =
+  let base, ext = Path.of_string path |> Path.split_extension in
+  let suffix = f ext in
+  Path.extend_basename base ~suffix |> Path.to_string
+
+let pp_with_ocamlc sctx pp_ml =
+  let ocamlc = (Super_context.context sctx).ocamlc |> Path.to_string in
+  let env = Super_context.context_env sctx |> Env.to_unix |> Array.of_list in
+  let argv =
+    Array.of_list [ocamlc; "-stop-after"; "parsing"; "-dsource"; pp_ml; "-dump-into-file"]
+  in
+  let pid =
+    Unix.create_process_env ocamlc argv env Unix.stdin Unix.stdout Unix.stderr
```

Comment:
Wouldn't it be better to use the `Process` module to handle process creation?

Reply:
Now it uses `Process.run` instead.